hydrolib/core/io/inifield/models.py

```diff
@@ -0,0 +1,244 @@
+            alias="operand",
+        )
+        averagingtype: Optional[str] = Field(
+            "Type of averaging, if interpolationMethod=averaging .",
+            alias="averagingtype",
+        )
+        averagingrelsize: Optional[str] = Field(
+            "Relative search cell size for averaging.", alias="averagingrelsize"
+        )
+        averagingnummin: Optional[str] = Field(
+            "Minimum number of points in averaging. Must be ≥ 1.",
+            alias="averagingnummin",
+        )
+        averagingpercentile: Optional[str] = Field(
+            "Percentile value for which data values to include in averaging. 0.0 means off.",
+            alias="averagingpercentile",
+        )
+        extrapolationmethod: Optional[str] = Field(
+            "Option for (spatial) extrapolation.", alias="extrapolationmethod"
+        )
+        locationtype: Optional[str] = Field(
+            "Target location of interpolation.", alias="locationtype"
+        )
+        value: Optional[str] = Field(
+            "Only for dataFileType=polygon. The constant value to be set inside for all model points inside the polygon."
+        )
+
+    comments: Comments = Comments()
+
+    quantity: str = Field(alias="quantity")
+    datafile: Path = Field(alias="dataFile")
+    datafiletype: DataFileType = Field(alias="dataFileType")
+    interpolationmethod: Optional[InterpolationMethod] = Field(
+        alias="interpolationMethod"
+    )
+    operand: Optional[Operand] = Field(Operand.override, alias="operand")
+    averagingtype: Optional[AveragingType] = Field(
+        AveragingType.mean, alias="averagingType"
+    )
+    averagingrelsize: Optional[NonNegativeFloat] = Field(1.01, alias="averagingRelSize")
+    averagingnummin: Optional[PositiveInt] = Field(1, alias="averagingNumMin")
+    averagingpercentile: Optional[NonNegativeFloat] = Field(
+        0, alias="averagingPercentile"
+    )
+    extrapolationmethod: Optional[bool] = Field(False, alias="extrapolationMethod")
+    locationtype: Optional[LocationType] = Field(LocationType.all, alias="locationType")
+    value: Optional[float] = Field(alias="value")
+
+    datafiletype_validator = get_enum_validator("datafiletype", enum=DataFileType)
+    interpolationmethod_validator = get_enum_validator(
+        "interpolationmethod", enum=InterpolationMethod
+    )
+    operand_validator = get_enum_validator("operand", enum=Operand)
+    averagingtype_validator = get_enum_validator("averagingtype", enum=AveragingType)
+    locationtype_validator = get_enum_validator("locationtype", enum=LocationType)
+
+    @validator("value", always=True)
+    @classmethod
+    def _validate_value_and_filetype(cls, v, values: dict):
+        if v is not None and values.get("datafiletype") != DataFileType.polygon:
+            raise ValueError(
+                f"When value={v} is given, dataFileType={DataFileType.polygon} is required."
+            )
```

Comment:
Shouldnt it be the other way around?
When datafiletype == DataFileType.polygon, `value` should not be None. (that is how I interpret the manual)

Reply:
We should use both:
* when v is given, dataFileType cannot be != polygon (my present check),
  or in other words, when dataFileType != polygon then v should not be given.
* And also, when dataFileType==polygon, then v must be given. I will add that once #75 is merged to main.